### ISSUE TITLE
comm_style tiled and fix balance with rcb can result in wrong forces when using KOKKOS

### ISSUE THREAD

**Issue by Leimeroth** [lammps/lammps#4400]
**Summary**

Using comm_style tiled and fix balance with the rcb option lead to wrong forces compared to standard communication in  a nanoindentation simulation in a small area of the box.
In my test cases it only occured when using GPUs via KOKKOS. In tests without KOKKOS on CPUs there was no problem.
I did not test CPUs with KOKKOS.

**LAMMPS Version and Platform**

LAMMPS (29 Aug 2024) on Ubuntu 24.04 with 8 NVIDIA GPUs.

**Expected Behavior**

All forces on atoms in both simulations should be equal at least within some numerical accuracy in the first timestep.

**Actual Behavior**

The forces differ massively for atoms in a small area in the box.

**Steps to Reproduce**
Run the lammps simulation with the provided input files.
Remove
```
comm_style tiled
```
and
```
fix rebalance all balance 1000 1.1 rcb
```
the input script and compare to output with the options.
The lammps run command I used is:
```
srun -n $SLURM_NTASKS lmp -k on g $SLURM_NTASKS -sf kk -pk kokkos newton on neigh half cuda/aware on -in control.inp
```
where $SLURM_NTASKS is 8.

**Further Information, Files, and Links**

[inputs.zip](https://github.com/user-attachments/files/18020123/inputs.zip)

**Comment by stanmoore1** [comment]
@Leimeroth thanks for the bug report, I can reproduce buggy behavior with input file you provided. Debugging now.

**Comment by stanmoore1** [comment]
@Leimeroth this should be fixed by [8cf03b0](https://github.com/lammps/lammps/pull/4399/commits/8cf03b0b1c12a05678dab6429348155641931e16) in https://github.com/lammps/lammps/pull/4399. Let us know if you see more issues.

**Comment by Leimeroth** [comment]
@stanmoore1 as far as I can tell there are no problems with the simulation anymore. Thanks a lot for the quick fix, it gives me a speedup of roughly 50% compared to the standard comm style with a slightly changed geometry compared to the input here.
